cocotb/share/lib/embed/gpi_embed.c

```diff
@@ -51,6 +54,68 @@ static char *argv[] = { progname };
 
 static PyObject *pEventFn = NULL;
 
+
+/**
+ * @name    handle_venv
+ * @brief   Set 
+ * @ingroup python_c_api
+ *
+ * GILState before calling: N/A
+ *
+ * GILState after calling: N/A
+ *
+ * Checks if running in Python virtualenv and if so,
+ * calls Py_SetProgramName with correct path to interpreter
+ */
+
+void handle_venv(void)
+{
+    static char venv_path[PATH_MAX];
+#if PY_MAJOR_VERSION >= 3
+    static wchar_t venv_path_w[PATH_MAX];
+#endif
+
+    char *venv_path_home = getenv("VIRTUAL_ENV");
+    if (venv_path_home) {
+        strncpy(venv_path, venv_path_home, sizeof(venv_path));
+        if (venv_path[sizeof(venv_path) - 1])
+            goto venv_err;
```

Comment:
You can simply do a `LOG_*` call here (ideally with a useful error message), and `return`. No need for the goto. (The same applies to other uses of `goto` in this function.)